lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -433,127 +451,21 @@ hashref from an array of L<Net::Async::Blockchain::Transaction>
 
 =cut
 
-async sub _check_contract_transaction {
-    my ($self, $transaction) = @_;
-
-    # ERC20 transfer
-    # transfer(address _to, uint256 _value)
-    # signature = 4 bytes
-    # address = 32 bytes
-    # amount = 32 bytes
-    # total = 68 bytes, characters = 136
-    # including "0x" = 138
-    if (length($transaction->data) >= 138 && substr($transaction->data, 0, 10) eq substr(TRANSFER_SIGNATURE, 0, 10)) {
-        my $address = substr($transaction->data, 10, 64);
-        my $amount  = substr($transaction->data, 74, 64);
-
-        return undef unless $address && $amount;
-
-        my $contract_address = $transaction->to;
-
-        my $hex_symbol = await $self->rpc_client->call({
-                data => SYMBOL_SIGNATURE,
-                to   => $contract_address
-            },
-            "latest"
-        );
+async sub _check_plugins {
+    my ($self, $transaction, $receipt) = @_;
 
-        my $symbol = $self->_to_string($hex_symbol);
-        return undef unless $symbol;
+    my @modules = $filter->modules();
+    my @transactions;
 
-        my $decimals = await $self->rpc_client->call({
-                data => DECIMALS_SIGNATURE,
-                to   => $contract_address
-            },
-            "latest"
-        );
-
-        if ($decimals) {
-            $transaction->{amount} = Math::BigFloat->from_hex($amount)->bdiv(Math::BigInt->new(10)->bpow($decimals))->bround(hex $decimals);
-        } else {
-            $transaction->{amount} = Math::BigFloat->from_hex($amount);
-        }
-
-        $transaction->{currency} = $symbol;
-        $transaction->{contract} = $contract_address;
-        $transaction->{to}       = $self->_remove_zeros($address);
-
-        return $transaction;
+    for my $module (grep { $_->can("enabled") && $_->enabled } @modules) {
```

Comment:
This `grep` seems to be better placed in the `filter` key upon constructing `$filter`, so you could just call `for my $module (@modules)` here...

Reply:
the modules method from Module::PluginFinder does not check the filter https://metacpan.org/release/Module-PluginFinder/source/lib/Module/PluginFinder.pm#L163-167

Reply:
Might be worth using `Module::Pluggable` here as @leonerd notes, and probably more straightfoward.  cf http://www.perladvent.org/2004/6th/